bitcoin/src/network/message_blockdata.rs

```diff
@@ -42,6 +42,22 @@ pub enum Inventory {
     },
 }
 
+impl Inventory {
+    /// Return the item value represented as a SHA256-d hash.
+    pub fn network_hash(&self) -> Option<[u8; 32]> {
+        match self {
+            Inventory::Error => None,
+            Inventory::Transaction(t) => Some(t.as_raw_hash().to_byte_array()),
```

Comment:
IIUC you shouldn't need `as_raw_hash`. (IOW, if you do, we messed up something.)

Reply:
Ah you're right. I basically just translated my old impl of `into_inner().to_bytes()` :)

Reply:
fixed